cilium/cmd/bpf_metrics_list.go

```diff
@@ -171,15 +171,17 @@ func extractRow(key, value string) (*metricsRow, error) {
 		return nil, fmt.Errorf("cannot extract reason and traffic direction from map's key \"%s\"", key)
 	}
 
-	reasonCode, err := strconv.Atoi(reasonCodeStr)
+	reasonCodeUint64, err := strconv.ParseUint(reasonCodeStr, 10, 8)
```

Comment:
Do we need to write the variable type in the name? We already know that `ParseUint` returns `uint64`s but in reality this is a `uint8` so maybe would be better to rename the suffix to `uint8` instead?

Reply:
`strconv.ParseUint` returns a `uint64` whose value should fit into a `uint8`, but there's no actual guarantee of this provided by Go's type system. The reason for adding the `Uint64` suffix to the `reasonCodeUint64`'s variable name is to make it explicit that this is a `reasonCode` (which is a `uint8` value) represented as a `uint64` value and to make the truncating typecast later explicit.

Note that the `reasonCodeUint64` variable has very limited scope, so this ugly variable name does not propagate far. Renaming `reasonCodeUint64` to `reasonCodeUint8` I think would cause confusion because (a) it's not a `uint8` and (b) it would make the truncating typecast be `uint8(reasonCodeUint8)` which looks strange: why convert what seems to already be a `uint8` to a `uint8`?

Does this reasoning make sense?

Reply:
It does yes, on one hand it's confusing to have prefixed `uint64` and on the other hand it's confusing to have prefixed `uint8`. I'm fine either way :slightly_smiling_face: